internal/pipe/mastodon/mastodon.go

```diff
@@ -0,0 +1,59 @@
+package mastodon
+
+import (
+	"fmt"
+
+	"github.com/caarlos0/env/v6"
+	"github.com/caarlos0/log"
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/context"
+	"github.com/mattn/go-mastodon"
+)
+
+const defaultMessageTemplate = `{{ .ProjectName }} {{ .Tag }} is out! Check it out at {{ .ReleaseURL }}`
+
+type Pipe struct{}
+
+func (Pipe) String() string                 { return "mastodon" }
+func (Pipe) Skip(ctx *context.Context) bool { return !ctx.Config.Announce.Mastodon.Enabled }
+
+type Config struct {
+	Server       string `env:"MASTODON_SERVER,notEmpty"`
```

Comment:
Would you rather have this in the announce config?

Reply:
I think the server could be there yes, maybe only the secrets could be in the env here

Reply:
I'll merge this and do it in a further commit!